CMakeLists.txt

```diff
@@ -102,15 +115,38 @@ set(TARGET_NAME slim)
 file(GLOB_RECURSE SLIM_SOURCES ${PROJECT_SOURCE_DIR}/core/*.cpp ${PROJECT_SOURCE_DIR}/eidos/*.cpp)
 add_executable(${TARGET_NAME} ${SLIM_SOURCES})
 target_include_directories(${TARGET_NAME} PRIVATE ${GSL_INCLUDES} "${PROJECT_SOURCE_DIR}/core" "${PROJECT_SOURCE_DIR}/eidos")
-target_link_libraries(${TARGET_NAME} PUBLIC gsl)
-target_link_libraries(${TARGET_NAME} PUBLIC tables)
-target_link_libraries(${TARGET_NAME} PUBLIC)
+target_link_libraries(${TARGET_NAME} PUBLIC gsl tables)
 
 set(TARGET_NAME eidos)
 file(GLOB_RECURSE EIDOS_SOURCES  ${PROJECT_SOURCE_DIR}/eidos/*.cpp  ${PROJECT_SOURCE_DIR}/eidostool/*.cpp)
 add_executable(${TARGET_NAME} ${EIDOS_SOURCES})
 target_include_directories(${TARGET_NAME} PRIVATE ${GSL_INCLUDES} "${PROJECT_SOURCE_DIR}/eidos")
-target_link_libraries(${TARGET_NAME} PUBLIC gsl)
-target_link_libraries(${TARGET_NAME} PUBLIC tables)
+target_link_libraries(${TARGET_NAME} PUBLIC gsl tables)
+
+set(TARGET_NAME eidoslib)
+file(GLOB_RECURSE EIDOS_SOURCES  ${PROJECT_SOURCE_DIR}/eidos/*.cpp  ${PROJECT_SOURCE_DIR}/eidostool/*.cpp)
+add_library(${TARGET_NAME} ${EIDOS_SOURCES})
+target_include_directories(${TARGET_NAME} PRIVATE ${GSL_INCLUDES} "${PROJECT_SOURCE_DIR}/eidos")
 
 install(TARGETS slim eidos DESTINATION bin)
+
+# QtSLiM -- this can be enabled with the -DBUILD_QTSLIM=ON option to cmake
+if(BUILD_QTSLIM)
+set(TARGET_NAME qtslim)
+find_package(Qt5 REQUIRED
+    Core
+    Gui
+    Widgets
+)
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+set(CMAKE_AUTOUIC ON)
+list(REMOVE_ITEM SLIM_SOURCES ${PROJECT_SOURCE_DIR}/core/main.cpp)
+file(GLOB_RECURSE QTSLIM_SOURCES ${PROJECT_SOURCE_DIR}/QtSLiM/*.cpp ${PROJECT_SOURCE_DIR}/QtSLiM/QtSLiMWindow.ui ${PROJECT_SOURCE_DIR}/QtSLiM/buttons.qrc ${PROJECT_SOURCE_DIR}/eidos/*.cpp)
+add_executable(${TARGET_NAME} "${QTSLIM_SOURCES}" "${SLIM_SOURCES}")
+set_target_properties( ${TARGET_NAME} PROPERTIES LINKER_LANGUAGE CXX)
+target_compile_definitions( ${TARGET_NAME} PRIVATE SLIMGUI=1 )
```

Comment:
Ah, so this raises a significant issue.  If you look at the current qmake files (*.pro), you can see that the files in slim and eidos need to be built with EIDOS_GUI, SLIMGUI=1, and SLIMPROFILING=0 all defined.  That is not optional; the code compiled when going into a GUI build actually needs to be different in various ways, to provide the hooks and privileges needed by the GUI to do its thing.  So I don't think the QtSLiM build will be able to rely on the same object files and libraries produced to build the command-line tools; they are fundamentally different builds.  Maybe you're handling that already – I'm not sure I'm following what you've done – but at a minimum those additional define symbols ought to be added here, for consistency with the builds using qmake and/or Xcode.

Reply:
That's good to know. I already define SLIMGUI. It's not too hard to just rebuild rather than reuse the libraries. The eidoslib is also not necessary in that case.

Reply:
The three flags are now defined and all sources are rebuilt for the qtslim target.